python/ribasim/ribasim/input_base.py

```diff
@@ -392,15 +393,24 @@ def check_parent(self) -> "ChildModel":
 class NodeModel(ChildModel):
     """Base class to handle combining the tables for a single node type."""
 
-    _sort_keys: dict[str, list[str]] = {}
-
     @model_serializer(mode="wrap")
     def set_modeld(
         self, serializer: Callable[[type["NodeModel"]], dict[str, Any]]
     ) -> dict[str, Any]:
         content = serializer(self)
         return dict(filter(lambda x: x[1], content.items()))
 
+    @field_validator("*")
+    @classmethod
+    def set_sort_keys(cls, v: Any, info: ValidationInfo) -> Any:
+        """Set sort keys for all TableModels if present in FieldInfo."""
+        if isinstance(v, (TableModel,)):
+            field = cls.model_fields[getattr(info, "field_name")]
+            extra = field.json_schema_extra
+            if extra is not None and isinstance(extra, dict):
```

Comment:
Is there a chance that extra is not a `dict`? Should we throw an assertion error in this case, because some programmer needs to do more work?

Reply:
It's to make mypy happy, as json_schema_extra can be a callable as well according to the type. We always set it to a dict though.